File: src/main/java/alfheim/common/integration/waila/WAILAAlfheimConfig.java
package alfheim.common.integration.waila;

import alfheim.common.block.tile.TileAnyavil;
import alfheim.common.block.tile.TileTradePortal;
import alfheim.common.integration.waila.handler.WAILAHandlerAnyavil;
import alfheim.common.integration.waila.handler.WAILAHandlerTradePortal;
import mcp.mobius.waila.api.impl.ModuleRegistrar;

public class WAILAAlfheimConfig {
	
	public static void loadConfig() {
		ModuleRegistrar.instance().registerBodyProvider(new WAILAHandlerAnyavil(), TileAnyavil.class);
		ModuleRegistrar.instance().registerNBTProvider(new WAILAHandlerAnyavil(), TileAnyavil.class);
		
		ModuleRegistrar.instance().registerBodyProvider(new WAILAHandlerTradePortal(), TileTradePortal.class);
		ModuleRegistrar.instance().registerNBTProvider(new WAILAHandlerTradePortal(), TileTradePortal.class);
	}
}

File: src/main/java/alfheim/common/integration/waila/WAILAAlfheimConfig.kt
package alfheim.common.integration.waila

import alfheim.common.block.tile.*
import mcp.mobius.waila.api.impl.ModuleRegistrar

object WAILAAlfheimConfig {
	
	fun loadConfig() {
		ModuleRegistrar.instance().registerBodyProvider(WAILAHandlerAnyavil(), TileAnyavil::class.java)
		ModuleRegistrar.instance().registerNBTProvider(WAILAHandlerAnyavil(), TileAnyavil::class.java)
		
		ModuleRegistrar.instance().registerBodyProvider(WAILAHandlerItemHolder(), TileManaAccelerator::class.java)
		ModuleRegistrar.instance().registerNBTProvider(WAILAHandlerItemHolder(), TileManaAccelerator::class.java)
		
		ModuleRegistrar.instance().registerBodyProvider(WAILAHandlerTradePortal(), TileTradePortal::class.java)
		ModuleRegistrar.instance().registerNBTProvider(WAILAHandlerTradePortal(), TileTradePortal::class.java)
	}
}
